uniffi_macros/src/export/trait_interface.rs

```diff
@@ -51,17 +51,17 @@ pub(super) fn gen_trait_scaffolding(
         /// Safety: Only pass pointers returned by a UniFFI call.  Do not pass pointers that were
         /// passed to the free function.
         pub unsafe extern "C" fn #clone_fn_ident(
-            ptr: *const ::std::ffi::c_void,
+            handle: ::std::primitive::u64,
             call_status: &mut ::uniffi::RustCallStatus
-        ) -> *const ::std::ffi::c_void {
-            ::uniffi::deps::trace!("clonining trait: {} ({:?})", #trait_name, ptr);
+        ) -> u64 {
+            ::uniffi::deps::trace!("clonining trait: {} ({:?})", #trait_name, handle);
             ::uniffi::rust_call(call_status, || {
-                let ptr = ptr as *mut ::std::sync::Arc<dyn #self_ident>;
+                let ptr = handle as *mut ::std::sync::Arc<dyn #self_ident>;
                 let arc: ::std::sync::Arc<_> = unsafe { ::std::clone::Clone::clone(&*ptr) };
                 ::std::result::Result::Ok(
                     ::std::boxed::Box::into_raw(
                         ::std::boxed::Box::new(arc),
-                    ) as *const ::std::ffi::c_void
+                    ) as u64
```

Comment:
```suggestion
                    ).expose_provenance() as u64
```

Same as with the other cast. I wish there was a nicer way of converting between u32 / u64 as well. Maybe adding an extension trait would actually be helpful 🤔